mshv-ioctls/src/ioctls/vm.rs

```diff
@@ -0,0 +1,263 @@
+        if ret == 0 {
+            Ok(())
+        } else {
+            Err(errno::Error::last())
+        }
+    }
+    ///
+    /// Creates/modifies a guest physical memory.
+    ///
+    pub fn map_user_memory(&self, user_memory_region: mshv_user_mem_region) -> Result<()> {
+        #[allow(clippy::cast_lossless)]
+        let ret = unsafe { ioctl_with_ref(self, MSHV_MAP_GUEST_MEMORY(), &user_memory_region) };
+        if ret == 0 {
+            Ok(())
+        } else {
+            Err(errno::Error::last())
+        }
+    }
+    ///
+    /// Unmap a guest physical memory.
+    ///
+    pub fn umap_user_memory(&self, user_memory_region: mshv_user_mem_region) -> Result<()> {
+        #[allow(clippy::cast_lossless)]
+        let ret = unsafe { ioctl_with_ref(self, MSHV_UNMAP_GUEST_MEMORY(), &user_memory_region) };
+        if ret == 0 {
+            Ok(())
+        } else {
+            Err(errno::Error::last())
+        }
+    }
+    ///
+    /// Creates a new MSHV vCPU file descriptor
+    ///
+    pub fn create_vcpu(&self, id: u8) -> Result<VcpuFd> {
+        let vp_arg = mshv_create_vp {
+            vp_index: id as __u32,
+        };
+        // Safe because we know that vm is a VM fd and we verify the return result.
+        #[allow(clippy::cast_lossless)]
+        let vcpu_fd = unsafe { ioctl_with_ref(&self.vm, MSHV_CREATE_VP(), &vp_arg) };
+        if vcpu_fd < 0 {
+            println!("vcpu fd: {}", vcpu_fd);
+            return Err(errno::Error::last());
+        }
+
+        // Wrap the vCPU now in case the following ? returns early. This is safe because we verified
+        // the value of the fd and we own the fd.
+        let vcpu = unsafe { File::from_raw_fd(vcpu_fd) };
+
+        Ok(new_vcpu(vcpu))
+    }
+    ///
+    /// Creates an interrupt controller.
+    ///
+    pub fn request_virtual_interrupt(
+        &self,
+        interrupt_type: hv_interrupt_type,
+        apic_id: u64,
+        vector: u32,
+        level_triggered: bool,
+        logical_destination_mode: bool,
+        long_mode: bool,
+    ) -> Result<()> {
+        let mut control_flags: u32 = 0;
+        if level_triggered {
+            control_flags |= 0x1;
+        }
+        if logical_destination_mode {
+            control_flags |= 0x2;
+        }
+        if long_mode {
+            control_flags |= 1 << 30;
+        }
+
+        let interrupt_arg = mshv_assert_interrupt {
+            control: hv_interrupt_control {
+                as_uint64: interrupt_type as u64 | ((control_flags as u64) << 32),
+            },
+            dest_addr: apic_id,
+            vector: vector,
+        };
+        #[allow(clippy::cast_lossless)]
+        let ret = unsafe { ioctl_with_ref(&self.vm, MSHV_ASSERT_INTERRUPT(), &interrupt_arg) };
+        if ret == 0 {
+            Ok(())
+        } else {
+            Err(errno::Error::last())
+        }
+    }
+    ///
+    /// Get a partition property
+    ///
```

Comment:
What is a partition property? Also what is the `code` argument we're passing here? Could you please document those?